master/pkg/searcher/asha.go

```diff
@@ -27,6 +27,7 @@ type (
 	}
 
 	asyncHalvingSearch struct {
+		defaultSearchMethod
```

Comment:
I would defer to @nrajanee  and @dzhu and @stoksc on the better design or experiment actor and search method but just want to point out:
if you make the search event queue an attribute of the experiment actor, instead of a search method, you will not have to add `defaultSearchMethod` parent to every existing search method.

Reply:
Yes, I agree that having it on the experiment actor would be odd because this is a property that is very specific to the custom searcher. I also did think about maybe having it as a searcher instead. We can talk about this more and I'll fix it in my next ticket. Does that sounds okay? defaultSearchMethod was used in randomSearch for example for default implementation of a few methods so I thought that would be appropriate in this scenario too.

Reply:
Leaving this as is, will revisit in next PR.